AdminCore.Services/EventService.cs

```diff
@@ -53,15 +53,20 @@ public IList<EventDto> GetEventsByEmployeeId(int employeeId, EventTypes eventTyp
       return _mapper.Map<IList<EventDto>>(QueryEventsByEmployeeId(eventTypeId, eventIds));
     }
 
-    public IList<EventDateDto> GetBookedEventDatesByEmployeeAndStartAndEndDatesAndEventStatus(DateTime startDate, DateTime endDate, int employeeId, EventStatuses eventStatuses)
-    {
-      var eventStatus = (int) eventStatuses;
-      var eventDates = DatabaseContext.EventDatesRepository.Get(x => (x.StartDate.Date >= startDate.Date
-                                                                         && x.EndDate.Date <= endDate.Date
-                                                                         || x.EndDate.Date == startDate.Date)
-                                                                         && x.Event.EmployeeId == employeeId
-                                                                         && x.Event.EventStatusId == eventStatus,
-                                                              null, x => x.Event);
+    public IList<EventDateDto> GetEventDatesByEmployeeAndStartAndEndDatesAndStatus(DateTime startDate, DateTime endDate, int employeeId, EventStatuses eventStatuses)
+    {
+      var eventStatus = (int)eventStatuses;
+      var eventDates = DatabaseContext.EventDatesRepository.Get(
+        current => 
```

Comment:
Just had a thought, should tests be written to cover all date scenarios to ensure that logic isn't missed?

```
Date        5   6   7   8
Booked  :   o           o
New     :   o       o

Booked  :   o           o
New     :       o   o

Booked  :   o           o
New     :       o       o

Booked  :   o           o
New     :   o           o

Booked  :       o   o
New     :   o           o
```

Reply:
Makes sense

Reply:
We could also update the IRepository for another method to check if something exists. 

Something like: TestRepository.Exists(x => x.id == 1) 
which returns a boolean